src/static/js/modules/jquery/cf_formValidator.js

```diff
@@ -0,0 +1,181 @@
+  },
+
+  // Validate an individual input for each of the set types
+  // @param   {object} $input The jQuery object of the input
+  // @param   {string} value  The evaluated value of the input
+  // @returns {object}        The status of each of the tested types
+  _validateInput: function( $input, value ) {
+    var status = {};
+
+    $.each( _validator.settings.types, function() {
+      var validation;
+      var type = this;
+      var notRequired = type === 'required' && !$input.prop( 'required' );
+      var notEmail = type === 'email' && $input.attr( 'type' ) !== 'email';
+      var notRadio = type === 'radiogroup' && $input.attr( 'type' ) !== 'radio';
+      var notCheckbox = type === 'checkgroup' && $input.attr( 'type' ) !== 'checkbox';
+
+      if ( notRequired || notEmail || notRadio || notCheckbox ) {
+        status[type] = null;
+        return false;
+      }
+
+      if ( type === 'required' ) {
+        validation = _validate.single( value, { presence: true } );
+      } else if ( type === 'email' ) {
+        validation = _validate.single( value, { email: true } );
+      }
+      status[type] = typeof validation === 'undefined';
+    } );
+
+    return status;
+  },
+
+  // Validate the fields of our form
+  // @returns {object} Two inner objects containing the valid and invalid fields
+  _validateFields: function( fields ) {
+    var checkgroups = {};
+    var validatedFields = {
+      valid:   [],
+      invalid: []
+    };
+
+    fields.each( function() {
+      var $input = $( this );
+      var failed = false;
+      var field;
+
+      // if it's a button elem or type is button or submit, skip it
+      if ( $input.is( 'button, :button, :submit' ) ) {
+        return;
+      // if it's a checkbox test the entire check group
+      } else if ( $input.is( ':checkbox' ) ) {
+        var name = $input.attr( 'name' );
+        var $group = $input.closest( '.form-group' );
+
+        if ( checkgroups[name] || !$group.hasClass( 'required-check-group' ) ) {
+          return;
+        }
+
+        var isChecked = $group.find( 'input:checked' ).length > 0;
+
+        field = {
+          elem:   $group,
+          value:  null,
+          label:  $.trim( $group.find( '.form-label-header' ).text() ),
+          status: {
+            checkgroup: isChecked
+          }
+        };
+
+        checkgroups[name] = true;
+      } else {
+        var value = $input.val();
+        var validation = _validator._validateInput( $input, value );
+
+        field = {
+          elem:   $input,
+          value:  value,
+          label:  $.trim( $( 'label[for="' + $input.attr( 'id' ) + '"]' ).text() ),
+          status: validation
+        };
+      }
+
+      for ( var prop in field.status ) {
+        if ( field.status[prop] === false ) {
+          failed = true;
+        }
+      }
+
+      if ( failed ) {
+        validatedFields.invalid.push( field );
+      } else {
+        validatedFields.valid.push( field );
+      }
```

Comment:
@anselmbradford you sure about that? https://github.com/cfpb/front-end/blob/master/javascript.md#bad-2